contracts/LinniaRecords.sol

```diff
@@ -83,6 +87,25 @@ contract LinniaRecords is Ownable, Pausable, Destructible {
         return true;
     }
 
+    /// @param dataHash the datahash of the record to be scored
+    /// @param irisProvidersAddress address of the oracle contract
+    function updateIris(bytes32 dataHash, address irisProvidersAddress)
+        external
+        onlyOwner
+        whenNotPaused
+    returns (uint)
+    {
+        require(irisProvidersAddress != address(0));
+        IrisScoreProviderI currOracle = IrisScoreProviderI(irisProvidersAddress);
+        uint val = currOracle.report(dataHash);
+        require(val > 0);
+
+        Record storage record = records[dataHash];
+        record.irisScore.add(val);
```

Comment:
Maybe a `hasNotReported` check for the irisProvider address if we increment the main iris score? A single provider could spam reports to inflate a file's iris score otherwise

Reply:
yep.  Good catch.

Reply:
updated